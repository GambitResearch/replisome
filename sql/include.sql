\set VERBOSITY terse

-- predictability
SET synchronous_commit = on;

DROP TABLE IF EXISTS t1;
DROP TABLE IF EXISTS t2;
DROP TABLE IF EXISTS t3;

CREATE TABLE t1 (id int PRIMARY KEY);
CREATE TABLE t2 (id int PRIMARY KEY);
CREATE TABLE t3 (id int PRIMARY KEY);
CREATE TABLE s1 (id int PRIMARY KEY);

SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');

insert into t1 values (1);
insert into t2 values (2);
insert into t3 values (3);
update t1 set id = 10 where id = 1;
update t2 set id = 20 where id = 2;
update t3 set id = 30 where id = 3;
delete from t1 where id = 10;
delete from t2 where id = 20;
delete from t3 where id = 30;

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'pretty-print', '1', 'include-table', 't1', 'include-table', 't3');


insert into t1 values (1);
insert into t2 values (2);
insert into s1 values (3);

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'pretty-print', '1', 'include-table', '~t');


insert into t1 values (4);
insert into t2 values (5);
insert into s1 values (6);

SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'pretty-print', '1', 'include-table', '~^.1$');


SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
