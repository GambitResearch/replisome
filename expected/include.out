\set VERBOSITY terse
-- predictability
SET synchronous_commit = on;
DROP TABLE IF EXISTS t1;
NOTICE:  table "t1" does not exist, skipping
DROP TABLE IF EXISTS t2;
NOTICE:  table "t2" does not exist, skipping
DROP TABLE IF EXISTS t3;
NOTICE:  table "t3" does not exist, skipping
CREATE TABLE t1 (id int PRIMARY KEY);
CREATE TABLE t2 (id int PRIMARY KEY);
CREATE TABLE t3 (id int PRIMARY KEY);
SELECT 'init' FROM pg_create_logical_replication_slot('regression_slot', 'wal2json');
 ?column? 
----------
 init
(1 row)

insert into t1 values (1);
insert into t2 values (2);
insert into t3 values (3);
update t1 set id = 10 where id = 1;
update t2 set id = 20 where id = 2;
update t3 set id = 30 where id = 3;
delete from t1 where id = 10;
delete from t2 where id = 20;
delete from t3 where id = 30;
SELECT data FROM pg_logical_slot_get_changes('regression_slot', NULL, NULL, 'include-xids', '0', 'pretty-print', '1', 'include-table', 't1', 'include-table', 't3');
                         data                          
-------------------------------------------------------
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "insert",            +
                         "schema": "public",          +
                         "table": "t1",               +
                         "columnnames": ["id"],       +
                         "columntypes": ["int4"],     +
                         "columnvalues": [1]          +
                 }
         ]                                            +
 }
 {                                                    +
         "change": [
         ]                                            +
 }
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "insert",            +
                         "schema": "public",          +
                         "table": "t3",               +
                         "columnnames": ["id"],       +
                         "columntypes": ["int4"],     +
                         "columnvalues": [3]          +
                 }
         ]                                            +
 }
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "update",            +
                         "schema": "public",          +
                         "table": "t1",               +
                         "columnnames": ["id"],       +
                         "columntypes": ["int4"],     +
                         "columnvalues": [10],        +
                         "oldkeys": {                 +
                                 "keynames": ["id"],  +
                                 "keytypes": ["int4"],+
                                 "keyvalues": [1]     +
                         }                            +
                 }
         ]                                            +
 }
 {                                                    +
         "change": [
         ]                                            +
 }
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "update",            +
                         "schema": "public",          +
                         "table": "t3",               +
                         "columnnames": ["id"],       +
                         "columntypes": ["int4"],     +
                         "columnvalues": [30],        +
                         "oldkeys": {                 +
                                 "keynames": ["id"],  +
                                 "keytypes": ["int4"],+
                                 "keyvalues": [3]     +
                         }                            +
                 }
         ]                                            +
 }
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "delete",            +
                         "schema": "public",          +
                         "table": "t1",               +
                         "oldkeys": {                 +
                                 "keynames": ["id"],  +
                                 "keytypes": ["int4"],+
                                 "keyvalues": [10]    +
                         }                            +
                 }
         ]                                            +
 }
 {                                                    +
         "change": [
         ]                                            +
 }
 {                                                    +
         "change": [
                 {                                    +
                         "kind": "delete",            +
                         "schema": "public",          +
                         "table": "t3",               +
                         "oldkeys": {                 +
                                 "keynames": ["id"],  +
                                 "keytypes": ["int4"],+
                                 "keyvalues": [30]    +
                         }                            +
                 }
         ]                                            +
 }
(24 rows)

SELECT 'stop' FROM pg_drop_replication_slot('regression_slot');
 ?column? 
----------
 stop
(1 row)

